src/types/test_result.go

```diff
@@ -0,0 +1,8 @@
+package types
+
+// TestResult represents the result of a test
+type TestResult struct {
+	Passed      bool
+	Interrupted bool
+	Error       error
```

Comment:
I don't like having an error inside of this and would prefer keeping the error separate as that is the common paradigm in go.